File: samples/client/petstore/go/StoreApi.go
package swagger;

import (
  "encoding/json"
  "fmt"
  "net/url"
)

type StoreApi struct { 
  basePath string
  apiInvoker ApiInvoker
}

func NewStoreApi() *obj{
    return &obj{basePath:"http://petstore.swagger.io/v2", apiInvoker:ApiInvoker.GetInstance()}
}

  
  /**
   * Returns pet inventories by status
   * Returns a map of status codes to quantities
   * @return Map[String]int32
   */
  func (response Map[String]int32) getInventory ()  {
    postBody := nil
    

    // create path and map variables
    path := "/store/inventory".Replace("\\{format\\}","json")
    

    queryParams := map[string]string {}
    headerParams := map[string]string {}
    formParams := map[string]string {}

    
    
    contentTypes := []string {  }

    
    

      response := apiInvoker.CallAPI(basePath, path, "GET", queryParams, postBody, headerParams, formParams, contentType)
      if response {
        return ApiInvoker.Deserialize(response, "map", "Map")
      } else {
        return 
      }
  }
  
  /**
   * Place an order for a pet
   * 
   * @param body order placed for purchasing the pet
   * @return Order
   */
  func (response Order) placeOrder (body )  {
    postBody := body
    

    // create path and map variables
    path := "/store/order".Replace("\\{format\\}","json")
    

    queryParams := map[string]string {}
    headerParams := map[string]string {}
    formParams := map[string]string {}

    
    
    contentTypes := []string {  }

    
    // body params
    _tempBody := nil
    if body {
      _tempBody = body
    }

      response := apiInvoker.CallAPI(basePath, path, "POST", queryParams, postBody, headerParams, formParams, contentType)
      if response {
        return ApiInvoker.Deserialize(response, "", "Order")
      } else {
        return 
      }
  }
  
  /**
   * Find purchase order by ID
   * For valid response try integer IDs with value &lt;= 5 or &gt; 10. Other values will generated exceptions
   * @param orderId ID of pet that needs to be fetched
   * @return Order
   */
  func (response Order) getOrderById (orderId )  {
    postBody := nil
    
    // verify the required parameter 'orderId' is set
    if orderId == nil {
       return 0, fmt.Error("Missing the required parameter 'orderId' when calling getOrderById")
    }
    

    // create path and map variables
    path := "/store/order/{orderId}".Replace("\\{format\\}","json")
    path = path.Replace("\\{" + "orderId" + "\\}", ApiInvoker.EscapeString(orderId))
    

    queryParams := map[string]string {}
    headerParams := map[string]string {}
    formParams := map[string]string {}

    
    
    contentTypes := []string {  }

    
    

      response := apiInvoker.CallAPI(basePath, path, "GET", queryParams, postBody, headerParams, formParams, contentType)
      if response {
        return ApiInvoker.Deserialize(response, "", "Order")
      } else {
        return 
      }
  }
  
  /**
   * Delete purchase order by ID
   * For valid response try integer IDs with value &lt; 1000. Anything above 1000 or nonintegers will generate API errors
   * @param orderId ID of the order that needs to be deleted
   * @return void
   */
  func (response ) deleteOrder (orderId )  {
    postBody := nil
    
    // verify the required parameter 'orderId' is set
    if orderId == nil {
       return 0, fmt.Error("Missing the required parameter 'orderId' when calling deleteOrder")
    }
    

    // create path and map variables
    path := "/store/order/{orderId}".Replace("\\{format\\}","json")
    path = path.Replace("\\{" + "orderId" + "\\}", ApiInvoker.EscapeString(orderId))
    

    queryParams := map[string]string {}
    headerParams := map[string]string {}
    formParams := map[string]string {}

    
    
    contentTypes := []string {  }

    
    

      response := apiInvoker.CallAPI(basePath, path, "DELETE", queryParams, postBody, headerParams, formParams, contentType)
      if response {
        return 
      } else {
        return nil
      }
  }
  

File: samples/server/petstore/grape/api/store_api.rb
require 'json'

module 
  class StoreApi
  
    resource :Store
    # summary: Returns pet inventories by status
    # description: 
      rescue_from :all do |e|
        Rack::Response.new([''], 200,  'Content-type' => 'text/error').finish
        
      end
      params do
        
      end 
      GET '/store/inventory'.gsub /{(.*?)}/, ':\1' do
  
      end
    end
  
    resource :Store
    # summary: Place an order for a pet
    # description: 
      rescue_from :all do |e|
        Rack::Response.new([''], 200,  'Content-type' => 'text/error').finish
        Rack::Response.new([''], 400,  'Content-type' => 'text/error').finish
        
      end
      params do
        optional :body, type: , desc: 'order placed for purchasing the pet'
        
      end 
      POST '/store/order'.gsub /{(.*?)}/, ':\1' do
  
      end
    end
  
    resource :Store
    # summary: Find purchase order by ID
    # description: 
      rescue_from :all do |e|
        Rack::Response.new([''], 404,  'Content-type' => 'text/error').finish
        Rack::Response.new([''], 200,  'Content-type' => 'text/error').finish
        Rack::Response.new([''], 400,  'Content-type' => 'text/error').finish
        
      end
      params do
        requires :orderId, type: , desc: 'ID of pet that needs to be fetched'
        
      end 
      GET '/store/order/{orderId}'.gsub /{(.*?)}/, ':\1' do
  
      end
    end
  
    resource :Store
    # summary: Delete purchase order by ID
    # description: 
      rescue_from :all do |e|
        Rack::Response.new([''], 404,  'Content-type' => 'text/error').finish
        Rack::Response.new([''], 400,  'Content-type' => 'text/error').finish
        
      end
      params do
        requires :orderId, type: , desc: 'ID of the order that needs to be deleted'
        
      end 
      DELETE '/store/order/{orderId}'.gsub /{(.*?)}/, ':\1' do
  
      end
    end
  
  end
end
